narwhals/dataframe.py

```diff
@@ -330,10 +330,24 @@ def unpivot(
         variable_name: str | None,
         value_name: str | None,
     ) -> Self:
+        on_ = [on] if isinstance(on, str) else on
+        index_ = (
+            [index]
+            if isinstance(index, str)
+            else index
+            if isinstance(index, list)
+            else []
+        )
+
+        if on_ is None:
+            on_ = [c for c in self.collect_schema().names() if c not in index_]
```

Comment:
@MarcoGorelli by pushing one level up, we need column names 🙃

Reply:
ah sorry i don't think we need to push this one, just the `str | list[str]` ones